server.js

```diff
@@ -1,33 +1,73 @@
-import express from 'express'
+import express, { response } from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import booksData from './data/books.json'
 
 // Defines the port the app will run on. Defaults to 8080, but can be 
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
 const port = process.env.PORT || 8080
-const app = express()
+const app = express() // a variable that allows us to create a server
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+// accepting get requests to this endpoint, / means no path after the address
+app.get('/', (request, response) => {
+  response.send('Welcome to my books API // Anna Hellqvist 📚')
+})
+
+// retrieves the whole books array
+app.get('/books', (request, response) => {
+
+  const { author, toplist, rating, language } = request.query;
+
+  if (author) {
+    const booksByAuthor = booksData.filter((book) => book.authors === author)
+    response.json(booksByAuthor);
+
+  } else if (toplist) {
+    const sortedBooks = [...booksData];
+    sortedBooks.sort((a, b) => b.average_rating - a.average_rating)
+    const topList = sortedBooks.slice(0, toplist);
+
+    response.json(topList);
+
+  } else if (rating) {
+    if (rating < 5) {
+      const filteredBooks = booksData.filter((book) => Math.floor(book.average_rating) === +rating);
+      response.json(filteredBooks)
+    } else {
+      response.json({ error: 'No books with that rating were found' })
+    }
+  } else if (language) {
+    const filteredLanguage = booksData.filter((book) => book.language_code === language);
+    response.json(filteredLanguage);
+  } else {
+    response.json(booksData);
+  }
+})
```

Comment:
Instead of having all these if/else, what do you think about refiltering the same data? Wouldn't it be nice if the user was able to query multiple things? 😇 Something like:

```
let filteredBooks
if (author) {
  filteredBooks  = //filtered on author
}
if (language) {
  filteredBooks = //filtered on language
}
res.json(filteredBooks)
```
and so on

Reply:
Also case insensitivity and partial query would be nice